### PR TITLE
Improve it

### PR DESCRIPTION
**Pull request by bjorn3**
* Decode benchmark output as utf-8 on error, instead of raw bytes.
* Build rustc_codegen_cranelift in release mode for fairer comparison to the other backends.
* Add some files to .gitignore
* `set -e`
* Fix template name
* Add missing `--clif` to `run_benchmark.py` invocation

### CODE REVIEW

**Comment by gabi-250** [comment]
@bjorn3 Thank you for the pull request, it is much appreciated! I am happy to merge it once the conflict is resolved.

**Comment by gabi-250** [comment]
@bjorn3 Thank you for the pull request, it is much appreciated! I am happy to merge this.

File: Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ ifeq ($(RUST_BACKEND),ironox)
 	CARGO := $(RUST_ROOT)/build/x86_64-unknown-linux-gnu/stage0/bin/cargo
 else ifeq ($(RUST_BACKEND),cranelift)
 	CLIF_SO := librustc_codegen_cranelift.so
-	BACKEND := $(realpath $(CLIF_DIR)/target/debug/$(CLIF_SO))
```
Comment (gabi-250) [inline Makefile:12]: Thank you for spotting this!

**Review by gabi-250 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (bjorn3) [reply Makefile:12]: It was actually correct before I changed `test.sh` to `./test.sh --release`.

Comment (gabi-250) [reply Makefile:12]: Yes, but I should've been running Cranelift in release mode.

**Pull request merged**